### PR TITLE
Extracted bits from #348

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
PR #348 has gotten stale, and in the mean time other PRs have implemented some of the same functions. To make it easier to rebase or reconstruct #348, I intend to split it up.

### CODE REVIEW

File: src/openloco/input.cpp [hunk 1]
```diff
@@ -51,9 +51,9 @@ namespace openloco::input
     }
 
     // 0x00406FEC
-    void enqueue_mouse_button(mouse_button button)
+    void enqueue_mouse_button(int32_t button)
     {
-        ((void (*)(int))0x00406FEC)((int32_t)button);
+        ((void (*)(int))0x00406FEC)(button);
```
Comment (duncanspumpkin) [inline src/openloco/input.cpp:56]: Are these regressions?

File: src/openloco/ui.cpp [hunk 2]
```diff
@@ -688,11 +688,11 @@ namespace openloco::ui
                     switch (e.button.button)
                     {
                         case SDL_BUTTON_LEFT:
-                            input::enqueue_mouse_button(mouse_button::left_pressed);
+                            input::enqueue_mouse_button(1);
```
Comment (duncanspumpkin) [inline src/openloco/ui.cpp:691]: Are these regressions?

File: src/openloco/ui/WindowManager.cpp [hunk 3]
```diff
@@ -1440,7 +1440,8 @@ namespace openloco::ui::WindowManager
                 continue;
 
             scrollIndex++;
-            if (window->scroll_areas[scrollIndex].flags & 0b10001)
+            constexpr uint16_t scrollbarFlags = scrollview::scroll_flags::HSCROLLBAR_VISIBLE | scrollview::scroll_flags::VSCROLLBAR_VISIBLE;
```
Comment (duncanspumpkin) [inline src/openloco/ui/WindowManager.cpp:1443]: Move into the header with the name `ANY_SCROLLBAR_VISIBLE`? Its used a in a few locations.

**Comment by duncanspumpkin** [comment]
Has this been rebased? A few of these changes look like regressions or have already been implemented.

File: src/openloco/input.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/openloco/input.cpp:56]: No, these changes were deliberate in eafbd687c773d2166ef9905f4140cb084be45f38. From the commit description:

> The pushed numbers are remapped to different values when popped. This explains why the pushed constants don't match their events (e.g. pushing `left_released` when right is pressed)

File: src/openloco/ui.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/openloco/ui.cpp:691]: See above

File: src/openloco/ui/WindowManager.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/openloco/ui/WindowManager.cpp:1443]: You mean as a mask? Sure, that sounds useful.

**Comment by AaronVanGeffen** [comment]
@duncanspumpkin These three commits were cherry-picked from #348 today, on top of then-current master. I think the changes will make more sense if you look at the individual commits.

**Comment by duncanspumpkin** [comment]
Okay drop 
5505dba its already implemented. Fine with the others since they have been done for a reason.

**Comment by AaronVanGeffen** [comment]
@duncanspumpkin Hmm, not quite. The commit is moving the headers from `window.h` to `scrollview.h`. I think the latter is more appropriate; `window.h` is big enough as it is…

**Comment by duncanspumpkin** [comment]
oh I'm sure i approved moving it in a previous PR woops.

**Pull request merged**